sklearn/metrics/pairwise.py

```diff
@@ -1277,6 +1280,8 @@ def pairwise_kernels(X, Y=None, metric="linear", filter_params=False,
             kwds = dict((k, kwds[k]) for k in kwds
                         if k in KERNEL_PARAMS[metric])
         func = PAIRWISE_KERNEL_FUNCTIONS[metric]
+    elif isinstance(metric, GPKernel):
```

Comment:
for compatibility between the pairwise module and GP's kernel module. You might pass a kernel instance as metric to pairwise_kernels(). I have to add a test for this.